Add module-info.java, fixes #2

zimmi opened the pull request:
I also took the liberty to do the following things, please tell me if I should revert some of them:

- Bump test dependency + plugin versions
- Set project.build.sourceEncoding to fix build warning
- Fix FileListTest when run as a module
- Bump Travis JDK version to oraclejdk9 (I assume that's the reason the build fails there right now)
- Set scm.url without property reference so it works on search.maven.org. You can see the current sad display here: https://search.maven.org/artifact/com.github.rzymek/opczip/1.2.0/jar

The `--add-opens` args for surefire are needed because JUnit reflectively calls the package private test methods, and that's not allowed when we are a module. An alternative would be to make every test class / method public, but that will probably just annoy future developers while providing little gain.

zimmi committed:
Add module-info.java, fixes #2

- Bump test dependency + plugin versions
- Set project.build.sourceEncoding to fix build warning
- Fix FileListTest when run as a module
- Bump Travis JDK version to oraclejdk9
- Set scm.url without property reference so it works on search.maven.org

rzymek commented on pom.xml:
```diff
@@ -32,15 +32,15 @@
-        <maven.compiler.target>1.8</maven.compiler.target>
-        <maven.compiler.source>1.8</maven.compiler.source>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <maven.compiler.release>9</maven.compiler.release>
```
I think we can achieve both Java 8 compatibility and Java 9+ module support with this:
https://maven.apache.org/plugins/maven-compiler-plugin/examples/module-info.html

rzymek replied:
We can achieve both Java 8 compatibility and Java 9+ modules using this: https://maven.apache.org/plugins/maven-compiler-plugin/examples/module-info.html

rzymek commented on src/test/java/com/github/rzymek/opczip/reader/FileListTest.java:
```diff
@@ -66,25 +69,15 @@ private InputStream open(String filename) {
-                false
-        );
+        try {
+            URI resource = FileListTest.class.getResource(XLSX_DIR).toURI();
```
:+1: Nice!

zimmi reviewed (commented):
Thanks for taking a look! The current code (Version 1.2.0) is already building with and using Java 9 features (namely Set::of, InputStream::transferTo, InputStream::readNBytes, InputStream::readAllBytes and Arrays::equals), so I'm not sure how to proceed. I see two possibilities:
1. Remove the Java 9 usage from the current code.
2. Make another PR against the last released version that was Java 8 (Version 1.0.2, that's also the version used by fastexcel). Then do a point release (1.0.3?), then merge the changes back into master.

rzymek commented:
Oh, damm! I haven't pushed my commit for Java 8 compatibility I've made a year ago :facepalm: 
Pushed them now.

zimmi commented:
Haha, always good to be prepared! I'll take a look.

zimmi commented:
@rzymek I rebased onto master and switched back to target Java 8. Please take another look. 

Note: The project still has to be compiled with Java 9+ so the compiler understands the module-info. If building with Java 8 is needed, we could use https://github.com/moditect/moditect (generates module-info.class by ASM magic).
